### PR TITLE
samplebrain: init at 0.18.5

### PR DESCRIPTION
**Pull request by mitchmindtree**
A custom sample mashing app designed by Aphex Twin, created by Dave Griffiths!

Recently shared by Warp Records [on twitter](https://twitter.com/WarpRecords/status/1573716350794698756?s=20&t=AyXKHfpELQMkasnJup3DIw).

Can confirm, it's good fun :)

Screenshot from 2022-09-25 22-06-25

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/samplebrain/default.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  lib,
+  liblo,
+  libsndfile,
+  makeDesktopItem,
+  mkDerivation,
+  portaudio,
+  qmake,
+  qtbase,
+  wrapQtAppsHook,
+}:
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/samplebrain/default.nix:13]:
```suggestion
{ lib
, fetchFromGitLab
, fftw
, liblo
, libsndfile
, makeDesktopItem
, portaudio
, qmake
, qtbase
, wrapQtAppsHook
}:

```

File: pkgs/applications/audio/samplebrain/default.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/samplebrain/default.nix:14]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/applications/audio/samplebrain/default.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    qmake
+    wrapQtAppsHook
+  ];
+  buildInputs = [
+    fftw
+    liblo
+    libsndfile
+    portaudio
+    qtbase
+  ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/samplebrain/default.nix:33]:
```suggestion
  version = "0.18";

  src = fetchFromGitLab {
    owner = "then-try-this";
    repo = "samplebrain";
    rev = "${version}_release";
    sha256 = "sha256-k5VFaIo6XKv9n6OQNNDLVneKQDBZA4eFu0RNea9DpVM=";
  };

  nativeBuildInputs = [
    qmake
    wrapQtAppsHook
  ];

  buildInputs = [
    fftw
    liblo
    libsndfile
    portaudio
    qtbase
  ];

```
Having some empty lines makes the code more readable and less one giant text block.

File: pkgs/applications/audio/samplebrain/default.nix [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  desktopItem = makeDesktopItem {
+    type = "Application";
+    desktopName = "${pname}";
+    name = "${pname}";
+    comment = "A sample masher designed by Aphex Twin";
+    exec = "${pname}";
+    icon = "${pname}";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/samplebrain/default.nix:40]:
```suggestion
    desktopName = pname;
    name = pname;
    comment = "A sample masher designed by Aphex Twin";
    exec = pname;
    icon = pname;
```

File: pkgs/applications/audio/samplebrain/default.nix [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    categories = [ "Audio" ];
+  };
+  installPhase = ''
+    mkdir -p $out/bin
+    cp samplebrain $out/bin
+    install -m 444 -D ${desktopItem}/share/applications/${pname}.desktop $out/share/applications/${pname}.desktop
+    install -m 444 -D desktop/${pname}.svg $out/share/icons/hicolor/scalable/apps/${pname}.svg
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/samplebrain/default.nix:48]:
```suggestion
  };

  installPhase = ''
    mkdir -p $out/bin
    cp samplebrain $out/bin
    install -m 444 -D desktop/samplebrain.svg $out/share/icons/hicolor/scalable/apps/samplebrain.svg
  '';
  
```
Please use copyDesktopItems to copy the generate desktop file(s)

File: pkgs/applications/audio/samplebrain/default.nix [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  meta = with lib; {
+    description = "A custom sample mashing app designed by Aphex Twin, made by Dave Griffiths";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/samplebrain/default.nix:50]:
```suggestion
    description = "A custom sample mashing app";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by mitchmindtree** [comment]
Comments have been addressed, and version updated from 0.18.0 -> 0.18.5.
cc @SuperSandro2000

**Pull request merged**